debian/snapd.prerm

```diff
@@ -0,0 +1,16 @@
+#! /bin/sh
+# prerm script for snapd
+
+set -e
+
+if [ "$1" = "remove" -o "$1" = "upgrade" ]; then
```

Comment:
Also, consider whether you mean just stop, or stop-and-disable, here (the latter is `disable --now`)